### PR TITLE
[edge-config] Fix SWR during development

### PR DESCRIPTION
**Pull request by AndyBitz**
Fixes a bug in the SWR functionality for Edge Config that is used during development.

At the moment, we don't consider the argument, causing the e.g. get function to always return the most recent promise. With this fix, we'll keep the entire Edge Config in memory and revalidate it only on a request in the background.

No Changeset to make it part of the 1.0.0 release.

### CODE REVIEW

File: packages/edge-config/src/index.common.test.ts [hunk 1]
```diff
@@ -389,3 +389,32 @@ describe('connectionStrings', () => {
...
+    const edgeConfig = pkg.createClient(
+      'https://edge-config.vercel.com/ecfg-2?token=token-2',
+    );
+    expect(await edgeConfig.get('foo')).toBe('bar');
+
+    fetchMock.mockResponse(JSON.stringify({ foo: 'bar2' }));
+    expect(await edgeConfig.get('foo')).toBe('bar'); // 1st call goes to the cache
+
+    await new Promise<void>((res) => {
+      setTimeout(res, 1000);
```
Comment (dferber90) [inline packages/edge-config/src/index.common.test.ts]: Does this really need to be 1s? This would slow the tests down right?

File: packages/edge-config/src/index.ts [hunk 2]
```diff
@@ -113,6 +112,70 @@ const getPrivateEdgeConfig = trace(
   },
 );
 
+/**
+ *
```
Comment (dferber90) [inline packages/edge-config/src/index.ts]: Description isn't saying a lot at the moment <picture data-single-emoji=":jjj_lol:" title=":jjj_lol:">:jjj_lol:</picture>

File: packages/edge-config/src/index.ts [hunk 3]
```diff
@@ -113,6 +112,70 @@ const getPrivateEdgeConfig = trace(
...
+ */
+function createGetInMemoryEdgeConfig(
+  useDevelopmentCache: boolean,
+  connection: Connection,
+  headers: Record<string, string>,
+): () => Promise<EmbeddedEdgeConfig | null> {
+  let embeddedEdgeConfigPromise: Promise<EmbeddedEdgeConfig | null> | null =
```
Comment (dferber90) [inline packages/edge-config/src/index.ts:127]: A description of embeddedEdgeConfigPromise and lastRequest would help make the code below more readable. At the moment I get it roughly but it's hard to follow some of the nuance

File: packages/edge-config/src/index.ts [hunk 4]
```diff
@@ -113,6 +112,70 @@ const getPrivateEdgeConfig = trace(
...
+    null;
+  let lastRequest: Promise<EmbeddedEdgeConfig | null> | null = null;
```
Comment (dferber90) [inline packages/edge-config/src/index.ts]: Should this be `latestRequest`?

File: packages/edge-config/src/index.ts [hunk 5]
```diff
@@ -208,17 +271,25 @@ export const createClient = trace(
      * While in development we use SWR-like behavior for the api client to
      * reduce latency.
      */
-    const shouldUseSwr =
+    const useDevelopmentCache =
```
Comment (dferber90) [inline packages/edge-config/src/index.ts]: We shouldn't prefix with "use" due to confusion with react hooks. Not sure it would actually lead to errors, but better to avoid this in the first place

```suggestion
    const shouldUseDevelopmentCache =
```

**Review by dferber90 — CHANGES_REQUESTED** [review]
So much simpler <picture data-single-emoji=":frog-10:" title=":frog-10:">:frog-10:</picture>

File: packages/edge-config/src/index.common.test.ts (hunk 1 above)
Comment (AndyBitz) [reply packages/edge-config/src/index.common.test.ts]: Not really, reduced it to 100ms.

It works even without any waiting, but less predictable, since we can't know when the promise gets resolved.

**Review by dferber90 — APPROVED** [review]
